LiteCore/Database/DatabaseImpl.cc

```diff
@@ -1040,10 +1047,8 @@ namespace litecore {
                 }
             }
 
-            if (remoteID > 0) {
-                // Found the remote ID!
-                return remoteID;
-            } else if (!canCreate) {
+            if (remoteID > 0 // Found the remote ID!
+                || !canCreate) {
```

Comment:
For the case of `!canCreate`, just curious why we don't break or return without even begin transaction. Is it because we want to try to check the existing remoteID for the second time?

Reply:
I may not understand your question. We do break without beginning transaction.

Reply:
I mean for the case that creating = true and canCreate = false.

Reply:
But as nobody is actually use canCreate option = false so I think preserve the current logic is totally fine.